packages/server/lib/models/data.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:enum_to_string/enum_to_string.dart';
+
+abstract class Data {
+  late DataType dataType;
+
+  Data(this.dataType);
+
+  Data.fromJson(Map<String, dynamic> json) {
+    var type = EnumToString.fromString(DataType.values, json['type']);
+    if (type == null)
+      dataType = DataType.Unknown;
+    else
+      dataType = type;
+  }
+
+  Map<String, dynamic> toJson() {
+    return {'type': EnumToString.convertToString(dataType)};
+  }
+}
+
+enum DataType { ServerInfo, UserInfo, Unknown, Error }
```

Comment:
Let's stick to convention used in project already - enums in lowercase DataType.serverInfo DataType.userInfo, DataType.unknown and so on